05/DB.test.js

```diff
@@ -35,52 +35,120 @@ describe(".insert()", () => {
 
         expect(insertedData).toEqual(data);
     });
+
+    it("reject when id is duplicated in the database", async () => {
+        const db = new DB();
+
+        // Wstawienie pierwszego rekordu
+        await db.insert({ id: 1, name: "devmentor" });
+
+        await expect(db.insert({ id: 1, name: "marcin" })).rejects.toEqual(
+            "ID can't be duplicated!"
+        );
+    });
+
+    it("inserts data when id is provided and not duplicated", async () => {
+        const db = new DB();
+        const data = { id: 1, name: "devmentor" };
+        const insertedData = await db.insert(data);
+
+        expect(insertedData).toEqual(data);
+    });
+
+    it("inserts data when id is not provided and generates a unique id", async () => {
+        const db = new DB();
+        const insertedData = await db.insert({ name: "devmentor" });
+
+        expect(insertedData.id).toBeDefined();
+    });
 });
+
 describe(".select()", () => {
-    it("ID not found", async () => {
-        expect.assertions(1);
+    it("resolves with the correct data when ID is found", async () => {
         const db = new DB();
-        await db.insert({ id: 1, name: "devmentor" });
-        const rows = await db.getRows();
+        const testData = { id: 1, name: "devmentor" };
+        await db.insert(testData);
 
-        expect(rows.length).toBe(1);
+        const selectedData = await db.select(1);
+
+        expect(selectedData).toEqual(testData);
+    });
+
+    it("rejects with 'ID not found' when ID is not found", async () => {
+        const db = new DB();
```

Comment:
Mamy warunkowe działania, bo "jeśli jest błąd to zrób to..." - w takim wypadku trzeba użyć `expect.assertions(1);` na samym początku